### PR TITLE
Enter an ECT's national insurance number to match a TRS record

### PR DESCRIPTION
**Pull request by TobyRet**
### Ticket

#512 

### Changes 

- Add a 'national insurance number' step to the Register ECT flow and direct the user to this step if TRS API cannot find a teacher with the submitted date of birth and TRN 
- Wire up national insurance number validator
- Direct the user to a 'not found' page if the TRS API can still not find a teacher with the submitted national insurance number
- Add national insurance numbers to test guidance component
- Refactoring: create a factory to handle storage for each step in the flow

#### Enter national insurance number

https://github.com/user-attachments/assets/9b407cfd-21f7-4645-b1a1-131e9ad57ba5

#### Teacher not found

https://github.com/user-attachments/assets/c11881d2-6107-410b-aab2-1154367d7f46

### CODE REVIEW

File: app/validators/national_insurance_number_validator.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+private
+
+  def error_message
+    options[:message] || I18n.t(national_insurance_number.error, scope: message_scope)
+  end
+
+  def message_scope
+    options[:message_scope] || DEFAULT_MESSAGE_SCOPE
+  end
```
Comment (tonyheadford) [inline app/validators/national_insurance_number_validator.rb:21]: Appreciate this was lifted from ECF1 but we probably want to remove those and use a string directly as I think we're avoiding i18n unless we have to

File: app/wizards/schools/register_ect/national_insurance_number_step.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module Schools
+  module RegisterECT
+    class NationalInsuranceNumberStep < StoredStep
+      attr_accessor :trn, :date_of_birth, :national_insurance_number
```
Comment (tonyheadford) [inline app/wizards/schools/register_ect/national_insurance_number_step.rb:6]: Should there be a `validates` in here somewhere for `:national_insurance_number` ?

Comment (TobyRet) [reply app/wizards/schools/register_ect/national_insurance_number_step.rb:6]: Yes! I've added it in and added missing unit tests

**Review by tonyheadford — APPROVED** [review]
Nice!

**Pull request merged**